### PR TITLE
HostObject Cleanup

### PR DESCRIPTION
**Pull request by BenVillalobos**
Cleanup PR post-https://github.com/dotnet/sdk-container-builds/pull/204

Each commit is a direct response to a comment/suggestion made on the previous PR.

### CODE REVIEW

**Comment by BenVillalobos** [comment]
This one's NOT marked for auto-merge 🙃

File: Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs [hunk 1]
```diff
@@ -176,12 +168,13 @@ protected override ProcessStartInfo GetProcessStartInfo
         string responseFileSwitch
     )
     {
+        HostObjectMagic();
```
Comment (rainersigwald) [inline Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs:171]: I would go further and inline it directly, deleting `HostObjectMagic` and making it all locals in this method.

File: Microsoft.NET.Build.Containers/Registry.cs [hunk 2]
```diff
@@ -232,8 +232,7 @@ public async Task Push(Image x, string name, string? tag, string baseName, Actio
 
         if (!putResponse2.IsSuccessStatusCode)
         {
-            string jsonResponse = await putResponse2.Content.ReadAsStringAsync();
-            throw new ContainerHttpException("Registry push failed.", putResponse2.RequestMessage?.RequestUri?.ToString(), jsonResponse);
+            throw new ContainerHttpException("Registry push failed.", putResponse2.RequestMessage?.RequestUri?.ToString(), jsonString);
```
Comment (baronfel) [inline Microsoft.NET.Build.Containers/Registry.cs:235]: the `jsonstring` here is the same as the `putResponse` - is that intended? I'd expect the content from `putResponse2` to go here.

Comment (BenVillalobos) [reply Microsoft.NET.Build.Containers/Registry.cs:235]: yeah, both responses use the same `manifestUploadContent`. in theory the only difference is the URI's they put to.

File: Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs [hunk 3]
```diff
@@ -176,12 +155,22 @@ protected override ProcessStartInfo GetProcessStartInfo
...
     )
     {
+        VSHostObject hostObj = new VSHostObject(HostObject as System.Collections.Generic.IEnumerable<ITaskItem>);
+        if (hostObj.ExtractCredentials(out string user, out string pass))
+        {
+            extractionInfo = (true, user, pass);
+        }
+        else
+        {
+            Log.LogWarning("Host object failed to extract");
```
Comment (rainersigwald) [inline Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs:165]: What keeps this from firing in the "there was no host object specified" case?

File: Microsoft.NET.Build.Containers/Registry.cs (hunk 2 above)
Comment (baronfel) [reply Microsoft.NET.Build.Containers/Registry.cs:235]: Thanks for the confirmation 👍

File: Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs (hunk 3 above)
Comment (BenVillalobos) [reply Microsoft.NET.Build.Containers/CreateNewImageToolTask.cs:165]: Good point. Logging a warning on false is a bit extreme for this. Turning it into a lower pri message.

**Review by baronfel — APPROVED** [review]
Thanks for coming back and polishing up those comments 👍

**Pull request merged**